forge/src/trace/identifier.rs

```diff
@@ -0,0 +1,75 @@
+use ethers::abi::{Abi, Address};
+use std::collections::BTreeMap;
+
+/// Trace identifiers figure out what ABIs and labels belong to all the addresses of the trace.
+pub trait TraceIdentifier {
```

Comment:
How would an EtherscanTraceIdentifier look like? If we have many addresses, we'd like to make all etherscan/4byte dir requests concurrently. Should the trait maybe have a `fn identify_addresses` function which would fetch Name/Label/ABI for all address/code pairs provided? This would allow us to join all the etherscan futures and block_on all of them at the same time (vs having to block in a loop).